tests/integration/test_task.py

```diff
@@ -80,3 +83,245 @@ def _run(self, input_files):
+    with pytest.raises(ValueError) as ex_err:
+        _py2ncl({"cow": 22}, None)
+    assert 'NCL does not support nested dicts:' in str(ex_err.value)
+
+
+def _get_single_base_task():
+    """Test BaseTask basic attributes."""
+    task = BaseTask(
+        name='task0',
+        ancestors=[BaseTask(name=f'task0-ancestor{j}') for j in range(2)],
+    )
+    for task0 in task.flatten():
+        task0.priority = 0
+    tasks = {task}
+
+    return tasks
+
+
+def test_base_task_names():
+    tasks = _get_single_base_task()
+    task_names = [task.name for task in tasks]
+    assert task_names == ['task0']
+    ancestor_names = [anc.name for anc in list(tasks)[0].ancestors]
+    assert ancestor_names == ['task0-ancestor0', 'task0-ancestor1']
+
+
+def test_individual_base_task_attrs():
+    tasks = _get_single_base_task()
+    task = list(tasks)[0]
+    assert task.products == set()
+    assert task.output_files is None
+    assert task.activity is None
+    assert task.priority == 0
+
+
+def _get_single_diagnostic_task(tmp_path, diag_script, write_diag=True):
+    """Assemble a simple DiagnosticTask object."""
+    diag_output_dir = tmp_path / 'mydiag'
+    diag_run_dir = diag_output_dir / 'run_dir'
+    diag_settings = {'run_dir': diag_run_dir, 'profile_diagnostic': False}
+    if write_diag:
+        with open(diag_script, "w") as fil:
+            fil.write("import os\n\nprint(os.getcwd())")
+
+    task = DiagnosticTask(
+        name='task0',
+        ancestors=[BaseTask(name=f'task0-ancestor{j}') for j in range(2)],
+        script=diag_script,
+        settings=diag_settings,
+        output_dir=diag_output_dir,
+    )
+    tasks = {task}
+
+    return tasks
+
+
+def test_py_diagnostic_task_basics(tmp_path):
+    """Test DiagnosticTask basic attributes."""
+    diag_script = tmp_path / 'diag_cow.py'
+    tasks = _get_single_diagnostic_task(tmp_path, diag_script)
+    task_names = [task.name for task in tasks]
+    assert task_names == ['task0']
+    ancestor_names = [anc.name for anc in list(tasks)[0].ancestors]
+    assert ancestor_names == ['task0-ancestor0', 'task0-ancestor1']
+    task = list(tasks)[0]
+    assert task.script == diag_script
+    assert task.settings == {'run_dir': tmp_path / 'mydiag' / 'run_dir',
+                             'profile_diagnostic': False}
+    assert task.output_dir == tmp_path / 'mydiag'
+
+
+def test_diagnostic_diag_script_none(tmp_path):
+    """Test case when diagnostic script doesnt exist."""
+    diag_script = tmp_path / 'diag_cow.py'
+    with pytest.raises(DiagnosticError) as err_msg:
+        _get_single_diagnostic_task(tmp_path, diag_script, write_diag=False)
+    diagnostics_root = os.path.join(DIAGNOSTICS_PATH, 'diag_scripts')
+    script_file = os.path.abspath(os.path.join(diagnostics_root, diag_script))
+    suf = "Cannot execute script "
+    ept = suf + "'{}' ({}): file does not exist.".format(script_file,
+                                                         script_file)
```

Comment:
bleh, can't be arsed - it's that I can't fit the whole message in one line and I don't want to use `\` :grin:

Reply:
You can use brackets to concatenate strings:
```python
>>> ('a' "b") == "ab"
True
```
instead of backslashes:
```python
    ept = (
        "Cannot execute script "
        "'{}' ({}): file does not exist.".format(script_file, script_file)
    )
```
but I think you could also just mention the `script_file` parameter only once?

Reply:
done with `()` nice one"!